include/aoscene.h

```diff
@@ -23,11 +54,37 @@ class AOScene : public QLabel {
   // no "combo" to speak of
   void combo_resize(int w, int h);
 
+  // Return the frame delay adjusted for speed
+  int get_frame_delay(int delay);
+
+  QStringList network_strings;
+
+  QString m_char;
+  QString m_emote;
+
 private:
-  QWidget *m_parent;
-  QMovie *m_movie;
   AOApplication *ao_app;
-  QString last_image;
+
+  QVector<QPixmap> movie_frames;
+  QVector<int> movie_delays;
+
+  // Effects such as sfx, screenshakes and realization flashes are stored in
+  // here. QString entry format: "sfx^[sfx_name]", "shake", "flash". The program
+  // uses the QVector index as reference.
+  QVector<QVector<QString>> movie_effects;
+
+  QTimer *preanim_timer;
+  QTimer *shout_timer;
+  QTimer *effect_timer;
+  QTimer *ticker;
+  QString last_path;
+  QImageReader *m_reader = new QImageReader();
+
+  QElapsedTimer actual_time;
+
+  // Usually used to turn seconds into milliseconds such as for [Time] tag in
+  // char.ini
+  const int time_mod = 60;
```

Comment:
In the documentation, this is the tick interval, so I would suggest renaming this to `tick_ms` or something like that.